### PR TITLE
optimize_for_inference_lib.fold_batch_norms() preserves data_format

### LINKED ISSUE #15034: Optimize graph & graph transform tools do not support NCHW
I tried optimizing graph using both [Graph transform tool](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/graph_transforms/README.md) and [Optimize graph for inference](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/tools/optimize_for_inference.py). Both cases produced the same error because the fused batchnorm used not NCHW, but NHWC. I've got the error like this:

```
InvalidArgumentError (see above for traceback): Must provide as many biases as the channel dimension of the input tensor: [256] vs. 19 in [1,256,19,19]
	 [[Node: prefix/convblock/BatchNorm/FusedBatchNorm = BiasAdd[T=DT_FLOAT, data_format="NHWC", _device="/job:localhost/replica:0/task:0/device:GPU:0"](prefix/convblock/Conv2D, prefix/convblock/Conv2D_bn_offset)]
```

Although NCHW is faster than NHWC in GPU environment, why the tools do not support NCHW?

### PR DESCRIPTION
**Pull request by yegord**
`fold_batch_norms()` currently breaks graphs containing convolutions using NCHW data format. The function replaces a BiasAdd operation with a new one, while not preserving the data format of the original operation. As a result, the new operation always has NHWC data format, and the execution of the resulting graph fails because of mismatching dimensions.

The proposed resolution is to copy the `data_format` property from the original operation.

The patch fixes https://github.com/tensorflow/tensorflow/issues/15034.

### CODE REVIEW

**Comment by qmick** [comment]
Could you please add a test for this fix that fails for master branch and pass for this PR code?

**Comment by yegord** [comment]
Signed.

**Comment by yegord** [comment]
@qmick Could you have a look at the added test?

**Comment by qmick** [comment]
Test looks fine. 

It is what we want? Ping @petewarden for the assignment of #15034, feel free to ignore this ping if it's not appropriate.

@yegord This PR should close #15034, right? It is strange that the keyword "fix #15034" did not ask for a close. Could you try to make it work?

**Comment by yegord** [comment]
Well, I have added the "Fixes NNN" line to the commit message.
GitHub should close the issue as soon as this commit is merged.
If you have other ideas about what I should do, please let me know.

**Comment by rmlarsen** [comment]
Could you please check if this change still needed? This might have been caused by a bug in graph optimization that is now fixed.

**Comment by yegord** [comment]
@rmlarsen The test from this PR still fails on master (55c1e4ae66ddb1866804c1d20ddc20aead12b392). In master `fold_batch_norm()` still replaces NCHW BiasAdd with NHWC BiasAdd. This still needs to be fixed. The change from this PR is still needed.

**Review by petewarden — APPROVED** [review]
